sw/otbn/code-snippets/randomness.s

```diff
@@ -111,11 +113,36 @@ test_urnd:
   /* Read the URND WSR again. */
   bn.wsrr w1, 0x2 /* URND */
 
-  /* Fail the test if the two numbers read from URND are equal. */
+  jal x0, error_checking
+
+/**
+ * This sobroutine performs the following verifications:
+ *
+ *  `w0 != w1` The two random number read shall be different.
+ *  `w0 != 0 ` The random number shall not have all bits equal to zero.
+ *  `w0 != ~0` The random number shall not have all bits equal to one.
+ */ 
+error_checking:
+  /* Fail the test if the two numbers read from RND are equal. */
   bn.cmp w0, w1
-  csrrs x2, 0x7c0 /* FG0 */, x0
-  andi x2, x2, 8 /* filter out Z (zero) bit */
-  addi x31, x0, 33
+  csrrs x2, CSR_FG0, x0
+  andi x2, x2, 8 /* Filter out Z (zero) bit.*/
```

Comment:
I know that this text was here before, but could we switch to something like "extract Z (zero) bit"? The problem is that "filter out" usually has the opposite meaning ("I filtered out all the grounds and was left with delicious coffee!")

Reply:
Thanks. Done!